lib/ecto/changeset.ex

```diff
@@ -1349,6 +1349,43 @@ defmodule Ecto.Changeset do
     end
   end
 
+  import Ecto.Query # probably not the right place to do this?
+  # @spec  ??
+  def unsafe_validate_unique(changeset, field_names, repo, error_field \\ :base, error_message \\ "not unique") do
```

Comment:
We also need some way to pass in a custom where clause as well.  For example, my email field has a unique index that uses a SQL function that is lowercasing it and trimming it and such, and since this is not using the uniqueness setup from the database itself then we need to be able to specify it manually here (still seems easily desyncable...).

Also, this really needs some marker saying `I AM SLOW BECAUSE I DO A DB QUERY`, unsure `unsafe` is quite accurate enough, but it is indeed unsafe regardless.  The repo variable may not be enough to signify this?